src/librustdoc/html/static/js/search.js

```diff
@@ -1308,22 +1322,29 @@ window.initSearch = function(rawSearchIndex) {
             return;
         }
 
+        var filterCrates = getFilterCrates();
+
+        // In case we have no information about the saved crate and there is a URL query parameter,
+        // we override it with the URL query parameter.
+        if (typeof filterCrates === "undefined" && params["filter-crate"] !== undefined) {
+            filterCrates = params["filter-crate"];
+        }
+
         // Update document title to maintain a meaningful browser history
         searchState.title = "Results for " + query.query + " - Rust";
 
         // Because searching is incremental by character, only the most
         // recent search query is added to the browser history.
         if (searchState.browserSupportsHistoryApi()) {
-            var newURL = getNakedUrl() + "?search=" + encodeURIComponent(query.raw) +
-                window.location.hash;
+            var newURL = buildUrl(query.raw, filterCrates);
+
             if (!history.state && !params.search) {
-                history.pushState(query, "", newURL);
+                history.pushState(query.raw, "", newURL);
```

Comment:
It was invalid before for some reason...

Reply:
Also, we don't use as it seems. So I guess it only makes it smaller?

Reply:
I'll clarify: We should just pass `null` here (and in the `replaceState` call below) because we never use the resulting state in our `popstate` event handler.

Reply:
Oh I see! Let's do this then!